Strata/DDM/Integration/Lean/HashCommands.lean

```diff
@@ -5,39 +5,108 @@
 -/
 
 import Strata.DDM.Integration.Lean.Env
-import Strata.DDM.Integration.Lean.Quote
 import Strata.DDM.Integration.Lean.ToExpr
 import Strata.DDM.TaggedRegions
 
 open Lean
-open Elab.Command (CommandElab CommandElabM elabCommand)
-open Elab.Term (TermElab)
-open Parser (InputContext)
+open Lean.Elab (throwUnsupportedSyntax)
+open Lean.Elab.Command (CommandElab CommandElabM)
+open Lean.Elab.Term (TermElab)
+open Lean.Parser (InputContext)
+open System (FilePath)
 
 namespace Strata
 
-class HasInputContext (m : Type → Type _) where
+class HasInputContext (m : Type → Type _) [Functor m] where
   getInputContext : m InputContext
+  getFileName : m FilePath :=
+    (fun ctx => FilePath.mk ctx.fileName) <$> getInputContext
```

Comment:
I'm not familiar enough with the syntax of lean to understand that. Can you please explain me what it means?